make.paws/tests/testthat/test_handlers_restxml.R

```diff
@@ -0,0 +1,692 @@
+        )
+      )
+  )
+  req <- new_request(svc, op_test, input, NULL)
+  req <- build(req)
+  r <- req$body
+  #<OperationRequest xmlns="https://foo/"><ListParam xmlns="https://foo/"><member xmlns="https://foo/">one</member><member xmlns="https://foo/">two</member><member xmlns="https://foo/">three</member></ListParam></OperationRequest>
+  expect_equal(r, xml_list_to_character(list(
+    OperationRequest = list(
+      ListParam = list(
+        member = list("one"),
+        member = list("two"),
+        member = list("three")
+      )
+    )
+  )))
+})
+
+test_that("NonFlattened List With LocationName Case1", {
+  op_test <- Operation(name = "OperationRequest")
+  op_input_test <- function(OperationRequest) {
+    args <- list(OperationRequest = OperationRequest)
+    interface <- structure(list(
+      OperationRequest = structure(list(
+          ListParam = structure(list(
+            structure(logical(0), type = "string")),
+            type = "list", locationName = "AlternateName",
+            locationNameList = "NotMember")
+      ), type = "structure", locationName = "OperationRequest")
+      ), type = "structure")
+    return(populate(args, interface))
+  }
+
+  input <- op_input_test(
+      OperationRequest = list(
+        ListParam = list(
+          "one",
+          "two",
+          "three"
+        )
+      )
+  )
+  req <- new_request(svc, op_test, input, NULL)
+  req <- build(req)
+  r <- req$body
+  expect_equal(r, xml_list_to_character(list(
+    OperationRequest = list(
+      AlternateName = list(
+        NotMember = list("one"),
+        NotMember = list("two"),
+        NotMember = list("three")
+      )
+    )
+  )))
+})
+
+test_that("Flattened List Case1", {
+  op_test <- Operation(name = "OperationRequest")
+  op_input_test <- function(OperationRequest) {
+    args <- list(OperationRequest = OperationRequest)
+    interface <- structure(list(
+      OperationRequest = structure(list(
+          ListParam = structure(list(
+            structure(logical(0), type = "string")),
+            type = "list", flattened = "true")
+      ), type = "structure", locationName = "OperationRequest")
+      ), type = "structure")
+    return(populate(args, interface))
+  }
+
+  input <- op_input_test(
+      OperationRequest = list(
+        ListParam = list(
+          "one",
+          "two",
+          "three"
+        )
+      )
+  )
+  req <- new_request(svc, op_test, input, NULL)
+  req <- build(req)
+  r <- req$body
+  expect_equal(r, xml_list_to_character(list(
+    OperationRequest = list(
+      ListParam = list("one"),
+      ListParam = list("two"),
+      ListParam = list("three")
+    )
+  )))
+})
+
+test_that("List of Structures Case1", {
+  op_test <- Operation(name = "OperationName")
+  op_input_test <- function(OperationRequest) {
+    args <- list(OperationRequest = OperationRequest)
+    interface <- structure(list(
+      OperationRequest = structure(list(
+          ListParam = structure(list(
```

Comment:
A list of structures should look like the following. The list shouldn't have any named elements itself.
```r
list(
  ListParam = list(
    list(
      Foo = 1,
      Bar = 2
    )
  )
)
```